### PR TITLE
feat: add mc get

### PR DESCRIPTION
**Pull request by jiuker**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description

add mc get

## How to test this PR?

1.deploy minio-server
2.make bucket
3.one shell for `mc admin trace minio| grep s3.`
4.`mc get alias/bucket/dir/object localdir/objectName` `mc get alias/bucket/dir/object localdir/`
5.You will found that only have `s3.GetObject` call.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fmc%2fpull%2fNNNNN)

### CODE REVIEW

**Comment by shtripat** [comment]
Am I missing something?
I see below two calls while running `./mc get m1/test-bucket/hosts ./abc/hosts`
```
2024-02-20T18:49:43.710 [200 OK] s3.GetBucketLocation localhost:9000/test-bucket/?location=  [::1]            163µs       ⇣  154.365µs  ↑ 93 B ↓ 128 B
2024-02-20T18:49:43.710 [200 OK] s3.GetObject localhost:9000/test-bucket/hosts [::1]            477µs       ⇣  463.146µs  ↑ 93 B ↓ 1.0 KiB
```

**Comment by jiuker** [comment]
> Am I missing something? I see below two calls while running `./mc get m1/test-bucket/hosts ./abc/hosts`
> 
> ```
> 2024-02-20T18:49:43.710 [200 OK] s3.GetBucketLocation localhost:9000/test-bucket/?location=  [::1]            163µs       ⇣  154.365µs  ↑ 93 B ↓ 128 B
> 2024-02-20T18:49:43.710 [200 OK] s3.GetObject localhost:9000/test-bucket/hosts [::1]            477µs       ⇣  463.146µs  ↑ 93 B ↓ 1.0 KiB
> ```

You should set env with `MC_REGION = Value`. @shtripat

File: cmd/get-main.go [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+// get command flags.
+var (
+	getFlags = []cli.Flag{}
+)
+
+// Get command.
+var getCmd = cli.Command{
+	Name:         "get",
+	Usage:        "get s3 object storage to local",
```
Comment (shtripat) [inline cmd/get-main.go:36]:
```suggestion
	Usage:        "get s3 object to local",
```

**Review by shtripat — APPROVED** [review]
lgtm

File: cmd/auto-complete.go [hunk 2]
```diff
@@ -508,6 +508,7 @@ var completeCmds = map[string]complete.Predictor{
 	"/quota/info":  aliasCompleter,
 	"/quota/clear": aliasCompleter,
 	"/put":         complete.PredictOr(s3Completer, fsCompleter),
+  "/get":         complete.PredictOr(s3Completer, fsCompleter),
```
Comment (harshavardhana) [inline cmd/auto-complete.go:511]:
```suggestion
      "/get":         complete.PredictOr(s3Completer, fsCompleter),
```

File: cmd/auto-complete.go [hunk 3]
```diff
@@ -508,6 +508,7 @@ var completeCmds = map[string]complete.Predictor{
 	"/quota/info":  aliasCompleter,
 	"/quota/clear": aliasCompleter,
 	"/put":         complete.PredictOr(s3Completer, fsCompleter),
+      "/get":         complete.PredictOr(s3Completer, fsCompleter),
```
Comment (harshavardhana) [inline cmd/auto-complete.go:511]:
```suggestion
       "/get":         complete.PredictOr(s3Completer, fsCompleter),
```

File: cmd/auto-complete.go [hunk 4]
```diff
@@ -508,6 +508,7 @@ var completeCmds = map[string]complete.Predictor{
 	"/quota/info":  aliasCompleter,
 	"/quota/clear": aliasCompleter,
 	"/put":         complete.PredictOr(s3Completer, fsCompleter),
+       "/get":         complete.PredictOr(s3Completer, fsCompleter),
```
Comment (harshavardhana) [inline cmd/auto-complete.go:511]:
```suggestion
        "/get":         complete.PredictOr(s3Completer, fsCompleter),
```

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**